### PR TITLE
Stats revamp v2 views and visitors graph improvements

### LINKED ISSUE #16274: [android] Stats: Views and Vistors Graph
It is the new and challenging graph with separate graphs for **Views** and **Visitors** respectively with bezier curves and gradient shadow below the curve split into previous and current weeks.  
- Views graph in blue color
- Visitors graph in purple color
- A tool tip with a comparison
- A title and legend above 
- Info text with +/- % change below

### PR DESCRIPTION
**Pull request by ravishanker**
This PR fixes all the outstanding issues on Views and Visitors graph from the following PRs
- [Views and Visitors Card](https://github.com/wordpress-mobile/WordPress-Android/pull/16258)
- [Views and Visitors Graph](https://github.com/wordpress-mobile/WordPress-Android/pull/16327)

Issues Fixed:
- Tool tip (Marker) now morphs to top, bottom, left and right based on location of the data point within graph
- Tool tip only appears only for current week line data
- data point circle only appears along with marker
- Shows ticks for data points on xAxis
- dates on xAxis are formatted better
- Copy of information line is updated from "Your this week's views ..". to "Your views this week"
- Chips are updated to correct colour
- Refactor and cleaned up

Fixes #16274 

| Marker | Marker | Marker
|---|---|---|
| Screenshot_20220517_001839|Screenshot_20220517_001734| Screenshot_20220517_001753|

To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab as in the images above (scroll down if necessary)

Test 2:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Notice that the Views graph is in blue and this week and previous week and percent change are all in order
- Notice the graph, data and change are all compare this week with previous week
- Switch to Visitors by tapping chip and notice graph is in purple color

Test 3:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Tap on the graph.  Notice a marker popup as in the third image above.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/LineChartLabelFormatter.kt [hunk 1]
```diff
@@ -1,17 +1,49 @@
 package org.wordpress.android.ui.stats.refresh.utils
 
+import android.nfc.FormatException
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/LineChartLabelFormatter.kt]: `nfc.FormatException` is not right I guess.

**Review by irfano — CHANGES_REQUESTED** [review]
This looks very good! Tooltip position works great!

- Added one small comment about `android.nfc.FormatException`
- Upper left corner of the tooltip is not rounded. 
- Dates are null on my emulator. It could be a problem with my local. We discussed that before. I'll work I'll work on that

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/LineChartLabelFormatter.kt (hunk 1 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/LineChartLabelFormatter.kt]: Good catch!  Imports sometimes go wrong.  Fixed
Also fixed left rounded corner and accessibility crash

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻 
Dates are null in my emulator. I'm adding this bug to the bug task in the project.

My only concern is accessibility. Talkback is not speaking anything when the line chart is selected.. Because I can't select any node on the line chart when talkback is active.
In the bar chart, I can select any day. Should we make improvements here, in this PR or a new PR, @ravishanker ? We can merge after deciding this.

**Pull request merged**